Utilities/DataCompression/include/DataCompression/DataDeflater.h

```diff
@@ -41,132 +64,130 @@ namespace AliceO2 {
    TODO: monitoring policy
    TODO: bit order: LSB to MSB in every byte or vice versa
  */
-
 template<
-  typename _RegType,
-  typename _TargetType,
-  //  typename BoundsCheck, // bounds check policy
-  class Codec
+  typename TargetType,
+  class Codec = CodecIdentity<TargetType>
   >
 class DataDeflater {
- public:
-  DataDeflater() : mBegin(nullptr), mEnd(nullptr), mCurrent(mEnd), mBitPosition(0), mCodec() {}
-  ~DataDeflater() {}
+public:
+  using target_type = TargetType;
+  static const std::size_t TargetBitWidth = 8 * sizeof(target_type);
+  using Writer = std::function<bool(const target_type&)>;
+
+  DataDeflater() : mCurrent(0), mFilledBits(0), mCodec() {}
+  ~DataDeflater() {
+    // check if the deflater is properly terminated, or pending data will be lost
+    assert(mFilledBits == 0);
+  }
 
   /**
-   * Init target
-   * TODO: think about other targets than a buffer
+   * reset deflater
```

Comment:
This comment does not help much: this is what I expect `DataDeflater::reset()` to do. How about indicating what are the cases this actually needs to be performed?